x/rollapp/keeper/msg_server_update_state.go

```diff
@@ -41,6 +43,10 @@ func (k msgServer) UpdateState(goCtx context.Context, msg *types.MsgUpdateState)
 				latestStateInfoIndex.Index, msg.RollappId)
 		}
 
+		// check if previous state last block desc has timestamp
+		lastBD := stateInfo.BDs.BD[len(stateInfo.BDs.BD)-1]
+		previousStateHasTimestamp = !lastBD.Timestamp.IsZero()
```

Comment:
this should be method on state info type

Reply:
addressed in https://github.com/dymensionxyz/dymension/pull/1098/commits/255d25142253343fdef550ae7c3c67f503370322

But todo:
https://github.com/dymensionxyz/dymension/blob/255d25142253343fdef550ae7c3c67f503370322/x/rollapp/types/state_info.go#L49-L53